lib/game.rb

```diff
@@ -0,0 +1,124 @@
+class Game
+
+  attr_accessor :board, :player_1, :player_2
+
+  WIN_COMBINATIONS =   [
+      [0,1,2],
+      [3,4,5],
+      [6,7,8],
+      [0,3,6],
+      [1,4,7],
+      [2,5,8],
+      [0,4,8],
+      [6,4,2]
+    ]
+
+  def initialize(player_1 = Players::Human.new("X"), player_2 = Players::Human.new("O"), board = Board.new)
+    @board = board
+    @player_1 = player_1
+    @player_2 = player_2
+  end
+
+  def start
+    puts "Welcome to Tic-tac-toe!"
+    puts "How many players?"
+    puts "Enter 0 to simulate a game between two computer players."
+    puts "Enter 1 for a game between a human player and computer player."
+    puts "Enter 2 for a 2 game between two human players."
+    puts "Enter WAR to simulate 100 computer vs. computer games."
+
+    input = gets.strip
+
+    game=Game.new
+
+    if input == "0"
+      game = Game.new(Players::Computer.new("X"), Players::Computer.new("O"))
+      game.play
+    elsif input == "1"
+      game = Game.new(Players::Human.new("X"), Players::Computer.new("O"))
+      game.play
+    elsif input == "2"
+      game.play  # two human players is the default
+    elsif input.upcase == "WAR"
+      draws = 0
+      100.times do
+        game = Game.new(Players::Computer.new("X"), Players::Computer.new("O"))
+        game.play
+        draws +=1 if game.draw?
+          end
+      puts "Out of 100 simulated games, there were #{draws} draws!"
+    else
+      puts "Invalid Input.  Please try again."
+      game.start
+    end
+
+    puts "Would you like to play again? y/n?"
+          input = gets.strip.downcase
+          if input == "y"
+            game.start
+          else
+            puts "Goodbye!"
+            exit
+          end
+  end
+
+  def current_player
+    @board.turn_count.even? ? @player_1 : @player_2
+  end
+
+  def opponent
+    @board.turn_count.odd? ? @player_1 : @player_2
+  end
+
+  def won?
+    WIN_COMBINATIONS.find do |combo|
+      if @board.cells[combo[0]] != " " && @board.cells[combo[0]] == @board.cells[combo[1]] && @board.cells[combo[1]] == @board.cells[combo[2]]
+        combo
+      else
+        false
+      end
+    end
+  end
+
+  def draw?
+    @board.full? && !won?
+  end
+
+  def over?
+     won? || draw?
+   end
+
+   def winner
+     if !won? || draw?
+       nil
+     else
+       @board.cells[won?[0]]
+     end
+   end
```

Comment:
Instead of calling `Game#won?` twice, consider using an if statement assignment. For example:

```ruby
if won = won?
  board.cells[won.first]
end
```

If `won?` returns true, then and only then will the local variable `won` be assigned, otherwise, the method will just abort with a false return value. But if `won` is assigned, we can use that variable instead of calling `won?` again within the if statement.